src/main/java/io/personium/common/auth/token/AbstractLocalToken.java

```diff
@@ -169,9 +174,18 @@ final String doCreateTokenString(final String[] extendedFields) {
         //
         String[] frag = tokenDecoded.split(SEPARATOR, -1);
 
-        // If wrong format, throw exception
+        // If the number of the fields is not as expected
+        if (frag.length != numFields) {
+            throw new TokenParseException(
```

Comment:
doParse method and populate method are used by many token classes. So changes effects on their classes. Please add all test code for parse method in all classes. 

All classes are:
GrantCode, PasswordChangeAccessToken, ResidentLocalAccessToken, ResidentRefreshToken, UnitLocalUnitUserToken, VisitorLocalAccessToken, and VisitorRefreshToken class.

Reply:
populate method gives arguments (token, issuer, numExtraFields + 7). But doCreateTokenString method makes 6 fields if schema is null and extraFields is null. Is this OK?

Reply:
Test for some tokens are covered in TokenTest
  - ResidentLocalAccessToken
  - ResidentRefreshToken
  - TransCellAccessToken
  - VisitorRefreshToken

Same kind of tests are missing for the following tokens.
  - GrantCode
  - PasswordChangeAccessToken
  - UnitLocalUnitUserToken
  - VisitorLocalAccessToken
So I will add these four

Reply:
Even if the schema / scope is null, doCreateTokenString will still put "" and tab separator .
So the minimum field length will always be 7.